crates/telio-firewall/src/firewall.rs

```diff
@@ -371,26 +384,32 @@ impl StatefullFirewall {
     ) -> bool {
         let ip = unwrap_option_or_return!(P::try_from(buffer), false);
         let peer: PublicKey = public_key.into();
-        {
-            // whitelist read-lock scope
-            let whitelist = unwrap_lock_or_return!(self.whitelist.read(), false);
-
-            // Fasttrack, if peer is whitelisted - skip any conntrack and allow immediately
-            if whitelist.peer_whitelist.contains(&peer) {
-                telio_log_trace!(
-                    "Outbound IP packet is for whitelisted peer, forwarding: {:?}",
-                    ip
-                );
-                return true;
+        let proto = ip.get_next_level_protocol();
+
+        // whitelist read-lock scope
+        let whitelist = unwrap_lock_or_return!(self.whitelist.read(), false);
+
+        // Fasttrack, if peer is whitelisted - skip any conntrack and allow immediately
+        if whitelist.peer_whitelist.contains(&peer) {
```

Comment:
```suggestion
if !feature_flag_enabled && whitelist.peer_whitelist.contains(&peer) {
    telio_log_trace!("Outbound IP packet is for whitelisted peer, forwarding: {:?}", ip);
    return true;
}

if !ip.check_valid() {
    telio_log_trace!("Outbound IP packet is not valid, dropping: {:?}", ip);
    return false;
}

match proto {
    IpNextHeaderProtocols::Udp => {
        self.handle_outbound_udp(&ip);
    }
    IpNextHeaderProtocols::Tcp => {
        self.handle_outbound_tcp(&ip);
    }
.............................
};

telio_log_trace!("Accepting packet {:?} {:?}", ip, peer);
true            
```

Reply:
Yeah, if we're going to make firewall aware of the feature flag then sure

Reply:
Actually, the thing is that you still need to allow packets for whitelisted peers even when the `check_valid()` fails. At least that it was before my changes

Reply:
Anyway, I can simply not call `handle_tcp_packet()` when the feature flag is off

Reply:
done